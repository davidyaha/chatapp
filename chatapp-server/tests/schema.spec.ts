import { Schema } from "./";
import { graphql } from "graphql";
import 'jest';

function assertNoError(res) {
    if ( res.errors ) {
        console.error(res.errors);
        expect(typeof res.errors).toBe("undefined");
    }
}

describe("Schema", () => {
    it("should export valid schema", () => {
        let query = Schema.getQueryType();
        expect(typeof query).toBe("object");

        let fields: any = query.getFields();
        expect(typeof fields).toBe("object");
    });

    it("should resolve testString correctly", () => {
        let testQuery = `{
            testString
        }`;

        let expectedResponse = {
            testString: "it Works!",
        };

        return graphql(Schema, testQuery, undefined, {}).then((res) => {
            assertNoError(res);
            expect(res.data).toMatchSnapshot();
        });
    });

    it("should resolve someType correctly", () => {
        let testQuery = `{
            someType {
                testFloat,
                testInt,
                fixedString,
            }
        }`;

        return graphql(Schema, testQuery, undefined, {}).then((res) => {
            assertNoError(res);
            expect(res.data).toMatchSnapshot();
        });
    });

    it("should resolve testStringConnector correctly", () => {
        let testQuery = `{
            testStringConnector
        }`;

        return graphql(Schema, testQuery, undefined, {}).then((res) => {
            assertNoError(res);
            expect(res.data).toMatchSnapshot();
        });
    });

    it("should resolve mockedObject  correctly", () => {
        let testQuery = `{
            mockedObject {
                mockedFirstName,
                mockedInt,
            }
        }`;

        return graphql(Schema, testQuery, undefined, {}).then((res) => {
            let data = res.data as {
              mockedObject: {
                mockedInt: number
                mockedFirstName: string
              }
            };

            assertNoError(res);
            expect(data.mockedObject.mockedInt).toBeGreaterThan(-1000);
            expect(data.mockedObject.mockedInt).toBeLessThan(1000);

            expect(data.mockedObject.mockedFirstName).toMatchSnapshot();
        });
    });
});
