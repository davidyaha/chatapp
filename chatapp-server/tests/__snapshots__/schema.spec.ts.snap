exports[`Schema should resolve mockedObject  correctly 1`] = `"Hello World"`;

exports[`Schema should resolve someType correctly 1`] = `
Object {
  "someType": Object {
    "fixedString": "fixed.",
    "testFloat": 303.0303,
    "testInt": 666,
  },
}
`;

exports[`Schema should resolve testString correctly 1`] = `
Object {
  "testString": "it Works!",
}
`;

exports[`Schema should resolve testStringConnector correctly 1`] = `
Object {
  "testStringConnector": "it works from connector as well!",
}
`;
